freeipa/src/test/java/com/sequenceiq/freeipa/service/upgrade/UpgradeServiceTest.java

```diff
@@ -98,6 +100,49 @@ public void testUpgradeTriggered() {
         assertEquals("pgw", upgradeEvent.getPrimareGwInstanceId());
         assertEquals(2, upgradeEvent.getInstanceIds().size());
         assertTrue(Set.of("im2", "im3").containsAll(upgradeEvent.getInstanceIds()));
+        assertFalse(upgradeEvent.isBackupSet());
+
+        verify(validationService).validateEntitlement(ACCOUNT_ID);
+        verify(validationService).validateStackForUpgrade(allInstances, stack);
+    }
+
+    @Test
+    public void testUpgradeTriggeredWithBackup() {
+        FreeIpaUpgradeRequest request = new FreeIpaUpgradeRequest();
+        request.setImage(new ImageSettingsRequest());
+        request.setEnvironmentCrn(ENVIRONMENT_CRN);
+
+        Stack stack = mock(Stack.class);
+        when(stack.getBackup()).thenReturn(new Backup());
+        when(stackService.getByEnvironmentCrnAndAccountIdWithLists(ENVIRONMENT_CRN, ACCOUNT_ID)).thenReturn(stack);
+        Set<InstanceMetaData> allInstances = createValidImSet();
+        when(stack.getNotDeletedInstanceMetaDataSet()).thenReturn(allInstances);
+        ImageInfoResponse selectedImage = new ImageInfoResponse();
+        when(imageService.selectImage(stack, request.getImage())).thenReturn(selectedImage);
+        ImageInfoResponse currentImage = new ImageInfoResponse();
+        when(imageService.currentImage(stack)).thenReturn(currentImage);
+        Operation operation = new Operation();
+        operation.setStatus(OperationState.RUNNING);
+        operation.setOperationId("opId");
+        when(operationService.startOperation(any(), any(), any(), any())).thenReturn(operation);
```

Comment:
I also might extract this, used in some other tests as well.

Reply:
ok